### PR TITLE
Moving to rxJava2

### PR DESCRIPTION
**Pull request by MattiaPrimavera**
Hi, here's a little summary of what it took,     

- update retrofit2 call factory for rx2
- moving rx.* to io.reactivex.*
- moving Subscription to Disposable
- moving testSubscriber to testObserver
- fixing RxAndroidSchedulersHook

Plus
-  0109961e4f0e833f5aac7d1b3b7c383f38061813: Updating gradle plugin version: 2.3.1 -> 2.3.3
- a661d169329709ce6cb433dbdcb0e22bb034a897: rxJava2 gradle imports added + fixing META-INF duplicate file issue 

Tests are all passing.

### CODE REVIEW

File: app/src/test/java/com/esoxjem/movieguide/util/RxSchedulersOverrideRule.java [hunk 1]
```diff
@@ -4,47 +4,18 @@
...
-import rx.plugins.RxJavaHooks;
-import rx.schedulers.Schedulers;
-
 /**
  * This rule registers SchedulerHooks for RxJava and RxAndroid to ensure that subscriptions
  * always subscribeOn and observeOn Schedulers.immediate().
  * Warning, this rule will reset RxAndroidPlugins and RxJavaPlugins before and after each test so
  * if the application code uses RxJava plugins this may affect the behaviour of the testing method.
  */
 public class RxSchedulersOverrideRule implements TestRule {
```
Comment (esoxjem) [inline app/src/test/java/com/esoxjem/movieguide/util/RxSchedulersOverrideRule.java]: Looks like this rule isn't required anymore? Please delete this class.

Comment (MattiaPrimavera) [reply app/src/test/java/com/esoxjem/movieguide/util/RxSchedulersOverrideRule.java]: Ok, I was waiting for confirmation on this, I ll do that

**Comment by MattiaPrimavera** [comment]
`shouldBeAbleToShowTrailers` and `shouldBeAbleToShowReviews` tests keeps failing `sometimes` only, it may be that those two are not run in a **deterministic way**, since the result is not always the same. Any idea why ?

**Comment by esoxjem** [comment]
@MattiaPrimavera The tests should pass as is, there's should't be any need to update them. Just change the `TestRule` to an RxJava2 rule instead. Something like [this](https://github.com/Plastix/RxSchedulerRule/blob/master/rx2/src/main/java/io/github/plastix/rxschedulerrule/RxSchedulerRule.java).

**Comment by MattiaPrimavera** [comment]
I followed [this article](https://medium.com/@fabioCollini/testing-asynchronous-rxjava-code-using-mockito-8ad831a16877) and played a little with the `@Rule`, but I did not really get why it does seem working with the `@Rule` **annotation** on every **PresenterImpl** but on the **MovieListing** one

**Pull request merged**